tutorials/messengers/telegram/4_conditions.py

```diff
@@ -58,70 +52,77 @@
 """
 
 
+# %%
+def check_if_latest_message_test_has_music(ctx: Context, _: Pipeline) -> bool:
+    if ctx.last_request is None:
+        return False
+    if ctx.last_request.original_message is None:
+        return False
+    if ctx.last_request.original_message.message is None:
+        return False
+    if ctx.last_request.original_message.message.text is None:
+        return False
+    return "music" in ctx.last_request.original_message.message.text
+
+
 # %%
 script = {
     "greeting_flow": {
         "start_node": {
             TRANSITIONS: {
-                "node1": telegram_condition(commands=["start", "restart"])
+                "node1": cnd.any(
+                    [
+                        cnd.exact_match(Message(text="/start")),
+                        cnd.exact_match(Message(text="/restart")),
+                    ]
+                )
             },
         },
         "node1": {
-            RESPONSE: TelegramMessage(text="Hi, how are you?"),
-            TRANSITIONS: {
-                "node2": telegram_condition(
-                    update_type=UpdateType.MESSAGE, regexp="fine"
-                )
-            },
-            # this is the same as
-            # TRANSITIONS: {"node2": telegram_condition(regexp="fine")},
+            RESPONSE: Message(text="Hi, how are you?"),
+            TRANSITIONS: {"node2": cnd.regexp("fine")},
         },
         "node2": {
-            RESPONSE: TelegramMessage(
-                text="Good. What do you want to talk about?"
-            ),
-            TRANSITIONS: {
-                "node3": telegram_condition(
-                    func=lambda msg: "music" in msg.text
-                )
-            },
+            RESPONSE: Message(text="Good. What do you want to talk about?"),
+            TRANSITIONS: {"node3": check_if_latest_message_test_has_music},
         },
         "node3": {
-            RESPONSE: TelegramMessage(
-                text="Sorry, I can not talk about music now."
-            ),
-            TRANSITIONS: {
-                "node4": telegram_condition(update_type=UpdateType.ALL)
-            },
+            RESPONSE: Message(text="Sorry, I can not talk about music now."),
+            TRANSITIONS: {"node4": lambda _, __, ___, ____: True},
             # This condition is true for any type of update
         },
         "node4": {
-            RESPONSE: TelegramMessage(text="bye"),
-            TRANSITIONS: {"node1": telegram_condition()},
+            RESPONSE: Message(text="bye"),
+            TRANSITIONS: {"node1": lambda _, __, ___, ____: True},
```

Comment:
No.

Don't need `*args, **kwargs`.

I doubt this script would even run successfully.